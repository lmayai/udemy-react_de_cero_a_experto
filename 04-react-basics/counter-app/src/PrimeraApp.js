// Los componentes se escriben con Upper Camel Case
// Functional Components
import React, { Fragment } from 'react';
import PropTypes from 'prop-types'

const PrimeraApp = ({mensaje, saludo = 'Hola Mundo' }) => {

  return (
    <Fragment>
      <h1> { saludo } </h1>
      {/* <pre>{JSON.stringify(variable, null, 3)}</pre> */}
      <p> { mensaje } </p>
    </Fragment>
  );
}

PrimeraApp.propTypes = {
  mensaje: PropTypes.string.isRequired,
  saludo: PropTypes.string,
}

export default PrimeraApp;
