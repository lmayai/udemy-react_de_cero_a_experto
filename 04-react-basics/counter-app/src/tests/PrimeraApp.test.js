import React from 'react'
import PrimeraApp from '../PrimeraApp';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

/* Para testing lybraru 
import { render } from '@testing-library/react';
*/

describe('Pruebas en <PrimeraApp />', () => {
  /* ESTA PRUEBA ES CON TESTING LIBRARY REACT
  
  test('should show the message "Primerita app"', () => {
    const saludo = 'Primerita app';
    const wrapper = render( <PrimeraApp mensaje={saludo} />);
    expect(wrapper.getByText(saludo)).toBeInTheDocument();
  
  })*/
  
  test('Should show <PrimerApp /> con Enzyme', () => {
    const saludo = 'Hola mundo';
    const wrapper = shallow(<PrimeraApp mensaje={saludo}/>);
    expect(wrapper).toMatchSnapshot();
  })

  test('should show the description, sent by props', () => {
    const saludo = 'Hola mundo';
    const description = 'Esta es la descripcion';
    const wrapper = shallow(
      <PrimeraApp 
        mensaje={saludo}
        description={description}
      />
    );

    // Esto es como un query selector
    //const textDescription = wrapper.find('small').html;
    const textDescription = wrapper.find('small').text();
    expect(textDescription).toBe(description);
  })
  
  


});
