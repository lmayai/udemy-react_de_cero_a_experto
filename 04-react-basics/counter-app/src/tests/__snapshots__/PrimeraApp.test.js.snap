// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> Should show <PrimerApp /> con Enzyme 1`] = `
<Fragment>
  <h1>
    Hola Mundo
  </h1>
  <p>
    Hola mundo
  </p>
  <small>
    Esta descripcion es por defecto
  </small>
</Fragment>
`;
